misk/src/main/kotlin/misk/config/MiskConfig.kt

```diff
@@ -58,20 +55,32 @@ object MiskConfig {
 
     val jsonNode = flattenYamlMap(configYamls)
 
-    @Suppress("UNCHECKED_CAST")
+    val configFile = "$appName-${environment.name.toLowerCase()}.yaml"
     try {
+      @Suppress("UNCHECKED_CAST")
       return mapper.readValue(jsonNode.toString(), configClass) as T
     } catch (e: MissingKotlinParameterException) {
-      val configFile = "$appName-${environment.name.toLowerCase()}.yaml"
       throw IllegalStateException(
           "could not find '${e.parameter.name}' of '${configClass.simpleName}'" +
               " in $configFile or in any of the combined logical config", e)
+    } catch (e: UnrecognizedPropertyException) {
+      val path = Joiner.on('.').join(e.path.map { it.fieldName })
+      throw IllegalStateException(
+          "error in $configFile: '$path' not found in '${configClass.simpleName}' ${suggestSpelling(e)}", e)
     } catch (e: Exception) {
       throw IllegalStateException(
           "failed to load configuration for $appName $environment: ${e.message}", e)
     }
   }
 
+  private fun suggestSpelling(e: UnrecognizedPropertyException): String {
+    val suggestions = e.knownPropertyIds.filter {
+      @Suppress("DEPRECATION")
+      StringUtils.getLevenshteinDistance(it.toString(), e.propertyName) <= 2
```

Comment:
this guy would pass pair programming interviews :)

Reply:
lol